src/main/java/org/roaringbitmap/ConcurrentBitmap.java

```diff
@@ -96,6 +105,53 @@ protected void set(final int x, final int bitValue) {
         }
     }
 
+    protected boolean checkedSet(final int x, final int bitValue) {
+        boolean check = false;
+        final short hb = Util.highbits(x);
+
+        Element e = highLowMap.get(hb);
+
+        if ((e == null) && (bitValue == 1)) {
+            // we have to create a new container for this x
+            check = true;
+            e = new Element(hb, new ArrayContainer());
+            Element existing = highLowMap.putIfAbsent(hb, e);
+            if (existing != null) {
+                e = existing;
+            }
+        }
+
+        // Element might still be null if we are trying to remove a bit from a
+        // non-existing container. We just ignore that request, otherwise:
+        if (e != null) {
+            short lb = Util.lowbits(x);
+            e.rwLock.writeLock().lock();
+            try {
+                Container container = e.value;
+                final Container afterUpdate;
+                if (bitValue == 1) {
+                    afterUpdate = container.add(lb);
+                    if (afterUpdate != container) {
+                        e.value = afterUpdate;
+                        check = true;
+                    }
+                } else {
+                    afterUpdate = container.remove(lb);
+                    if (container.getCardinality() == 0) {
+                        highLowMap.remove(hb);
+                        check = true;
+                    } else if (afterUpdate != container) {
```

Comment:
An element might be effectively added or removed, and the container remain the same. Containers only change when adding or removing lots of elements ( e.g. after adding 4096 elements an array container is converted into a bitmap container)